Objective-C/CBLDatabase.mm

```diff
@@ -178,12 +178,16 @@ - (NSString*) description {
 
 
 - (NSString*) path {
-    return _c4db != nullptr ? sliceResult2FilesystemPath(c4db_getPath(_c4db)) : nil;
+    @synchronized (self) {
```

Comment:
Question: Could `@synchronized (self)` be the best approach instead of `@synchronized (non-self variable)`.  Same questions for all `@synchronized (self)`